soccer/planning/PlannerNode.cpp

```diff
@@ -0,0 +1,147 @@
+#include "PlannerNode.hpp"
+#include "Robot.hpp"
+#include "planning/planner/CollectPlanner.hpp"
+#include "planning/planner/EscapeObstaclesPathPlanner.hpp"
+#include "planning/planner/LineKickPlanner.hpp"
+#include "planning/planner/PathTargetPlanner.hpp"
+#include "planning/planner/PivotPathPlanner.hpp"
+#include "planning/planner/SettlePlanner.hpp"
+
+namespace Planning {
+
+PlannerNode::PlannerNode(Context* context)
+    : context_(context), plannerIdx(Num_Shells, -1) {
+    planners_.push_back(std::make_unique<PathTargetPlanner>());
+    planners_.push_back(std::make_unique<SettlePlanner>());
+    planners_.push_back(std::make_unique<CollectPlanner>());
+    planners_.push_back(std::make_unique<LineKickPlanner>());
+    planners_.push_back(std::make_unique<PivotPathPlanner>());
```

Comment:
It feels weird to be treating these planners as state agnostic by having only one instantiation for each planner, then to be storing the state of each robot within the planners themselves. It adds a bunch of complexity in the code without adding anything? I'm only halfway through this PR so I might not have seen it yet.

Reply:
Yeah, I think you're right. I'd be fine with having a `vector<vector<unique_ptr<Planner>>>`